src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.utils.js

```diff
@@ -52,7 +52,20 @@ export const structureData = function (parentPath, dataForParentPath = []) {
     }
 };
 
-const adaptedRow = r => ({
+export const flattenTree = function (rows, selector = p => p) {
```

Comment:
just as alternative implementation - I would actually remove the selector part and just use map instead when using it: `flattenTree(rows).map(p => p)`, for example. This decouples the transformation part, esp that it's optional in some cases.